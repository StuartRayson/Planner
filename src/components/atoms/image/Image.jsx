import classNames from 'classnames';
import React from 'react';
import PropTypes from 'prop-types';

const Image = (props) => {
  const classes = classNames(
    'a-image',
    props.className
  );

  return (
    <img {...props} alt={props.alt} className={classes} />
  );
};

Image.propTypes = {
  alt: PropTypes.string.isRequired,
  className: PropTypes.string,
  src: PropTypes.string.isRequired,
  width: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number
  ]),
  height: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number
  ])
};

export default Image;
