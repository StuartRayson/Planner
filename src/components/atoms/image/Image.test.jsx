import React from 'react';
import { shallow } from 'enzyme';

import Image from './Image.jsx';

const imageSrc = '../../../shared/assets/images/logo.svg';
const ImageAlt = 'logo';

describe('<Image />', () => {
  it('renders without crashing', () => {
    shallow(<Image src={imageSrc} alt={ImageAlt} />);
  });

  it('should render with a src attribute', () => {
    const image = shallow(<Image src={imageSrc} alt={ImageAlt} />);
    expect(image.find('img').prop('src')).toEqual(imageSrc);
  });

  it('should render with a alt attribute', () => {
    const image = shallow(<Image src={imageSrc} alt={ImageAlt} />);
    expect(image.find('img').prop('alt')).toEqual(ImageAlt);
  });

  it('should render and match snapshot with basic config', () => {
    const wrapper = shallow(
      <Image src={imageSrc} alt={ImageAlt} />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
