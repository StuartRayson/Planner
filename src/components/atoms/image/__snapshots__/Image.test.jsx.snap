// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Image /> should render and match snapshot with basic config 1`] = `
<img
  alt="logo"
  className="a-image"
  src="../../../shared/assets/images/logo.svg"
/>
`;
